tendermint-rs/src/public_keys.rs

```diff
@@ -1,17 +1,39 @@
 //! Public keys used in Tendermint networks
 
-use crate::{amino_types::PubKeyResponse, error::Error};
+use crate::error::Error;
+#[cfg(feature = "serde")]
+use serde::{de::Error as DeError, Deserialize, Deserializer, Serialize, Serializer};
 use signatory::{ecdsa::curve::secp256k1, ed25519};
 use std::ops::Deref;
+#[cfg(feature = "serde")]
+use subtle_encoding::base64;
 use subtle_encoding::{bech32, hex};
 
 /// Public keys allowed in Tendermint protocols
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "type", content = "value"))]
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
 pub enum PublicKey {
     /// Ed25519 keys
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            rename = "tendermint/PubKeyEd25519",
+            serialize_with = "serialize_ed25519_base64",
+            deserialize_with = "deserialize_ed25519_base64"
+        )
+    )]
     Ed25519(ed25519::PublicKey),
 
     /// Secp256k1 keys
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            rename = "tendermint/PubKeySecp256k1",
+            serialize_with = "serialize_secp256k1_base64",
+            deserialize_with = "deserialize_secp256k1_base64"
```

Comment:
Haven't tested any of this but the names appear to match what's in Tendermint

Reply:
But https://github.com/tendermint/kms/pull/228#issuecomment-483374317 means you tested this manually?

Reply:
There's a test for Ed25519 keys here:

https://github.com/tendermint/kms/pull/228/files/611692b6455408c5f01ad50eedde9a13ec47dabc#diff-e73395ccad975224b12459de892a0722R217

I haven't tested anything secp256k1-related, however I'll add the account key test vectors at least. They won't exercise this, but it seems like if Ed25519 works then barring a spelling mistake above secp256k1 should work too